packages/flutter/test/widgets/actions_test.dart

```diff
@@ -965,7 +965,7 @@ void main() {
         expect(buttonNode2.hasFocus, isFalse);
         primaryFocus!.nextFocus();
         await tester.pump();
-        expect(buttonNode1.hasFocus, isTrue);
+        expect(buttonNode1.hasFocus, isFalse);
```

Comment:
I am not sure which one is more correct. Previously if every node is not traversable, pressing tab the focus will stay at the current focus.

After the change, it will unfocus the current focus.

I can preserve the previous behavior, but that will add a more complexity (for example i need to cache previous focused node and restore it if it can't find the a traversible node)

but I think this behavior probably doesn't worth restoring.

Reply:
Yes, I think it's fine to unfocus in that case. The only case I worry about is the trivial one where the tree only contains one focus node and the root scope, because then when it unfocuses, it will focus the root scope, which traps the focus so that even if the widget tree changes later, traversal won't work anymore, since the root scope doesn't have any geometry associated with it, so `findFirstFocus` can't find anything.  But the trivial case is pretty hard to create, since you'd have to have an app that doesn't use `WidgetsApp` as its base, and doesn't create any scopes of its own.

Reply:
Do you mean this `FocusManager.instance.rootScope`? I have a check that in the TraversalEdgeBehavior.parentScope to make sure it won't give up focus if the parent is `FocusManager.instance.rootScope`

Reply:
Yes, that's what I mean.  Sounds good.